dlt/pipeline/pipeline.py

```diff
@@ -1053,6 +1059,21 @@ def get_local_state_val(self, key: str) -> Any:
             state = self._get_state()
         return state["_local"][key]  # type: ignore
 
+    def _update_last_run_context(self) -> None:
+        """
+        Persist the directory context in local state.
+        Safe-no-op if the pipeline is not active or if run_context cannot be resolved.
+        """
+        try:
+            ctx = dlt.current.run_context()
```

Comment:
this must be always available. I plan to read this when we initialize the pipeline (in dlt.pipeline) and pass it to the class... tldr;> you do not need exception handler